### PR TITLE
[5.0] Fix tinyMCE upgrade hiccup

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue #41473

### Summary of Changes

The upgrade process could corrupt the data, thus a conditional was introduced

### Testing Instructions

Follow the issue, ie this comment: https://github.com/joomla/joomla-cms/issues/41473#issuecomment-1694337380

### Actual result BEFORE applying this Pull Request

`jjtemplate` instead of `jtemplate` in the db

### Expected result AFTER applying this Pull Request

works

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_admin/script.php [hunk 1]
```diff
@@ -2300,10 +2300,16 @@ private function migrateTinymceConfiguration(): bool
...
+                $replace = ['blocks', 'fontfamily', 'fontsize', 'styles'];
+
+                // Don't redo the template
+                if (!str_contains($params['configuration']['toolbars'][$setIdx]['menu'], 'jtemplate')) {
+                    $search[]  = 'template';
+                    $replace[] = 'jtemplate';
+                }
+
+                $params['configuration']['toolbars'][$setIdx]['menu'] = str_replace($search, $replace, $toolbarConfig['menu']
                 );
```
Comment (heelc29) [inline administrator/components/com_admin/script.php]:
```suggestion
                $params['configuration']['toolbars'][$setIdx]['menu'] = str_replace($search, $replace, $toolbarConfig['menu']);
```
drone fail (phpcs) so no download package is available

**Comment by heelc29** [comment]
I have tested this item :red_circle: unsuccessfully on a71240a03b5d8968406c292f25db18bf396c79cf<br /><br />image
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41475">issues.joomla.org/tracker/joomla-cms/41475</a>.</sub>

File: administrator/components/com_admin/script.php [hunk 2]
```diff
@@ -2319,11 +2324,16 @@ private function migrateTinymceConfiguration(): bool
...
-                    $params['configuration']['toolbars'][$setIdx][$toolbarIdx] = str_replace(
-                        ['fontselect', 'fontsizeselect', 'formatselect', 'styleselect', 'template'],
-                        ['fontfamily', 'fontsize', 'blocks', 'styles', 'jtemplate'],
-                        $toolbarConfig[$toolbarIdx]
-                    );
+                    $search  = ['fontselect', 'fontsizeselect', 'formatselect', 'styleselect'];
+                    $replace = ['fontfamily', 'fontsize', 'blocks', 'styles'];
+
+                    // Don't redo the template
+                    if (!in_array('jtemplate', $params['configuration']['toolbars'][$setIdx]['menu']))) {
```
Comment (richard67) [inline administrator/components/com_admin/script.php]:
```suggestion
                    if (!in_array('jtemplate', $params['configuration']['toolbars'][$setIdx][$toolbarIdx]))) {
```

**Comment by heelc29** [comment]
update from 5.0.0-alpha4 works
update from 4.4.0-alpha4 looks strange ... I repeat it
image

**Comment by heelc29** [comment]
I have tested this item :white_check_mark: successfully on 612e1349d18389f705a38556c169c5c8542a1eac<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41475">issues.joomla.org/tracker/joomla-cms/41475</a>.</sub>

**Comment by dgrammatiko** [comment]
@heelc29 could you paste the tinyMCE params column that produced this error?

TBH I have no clue what happens there...

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 612e1349d18389f705a38556c169c5c8542a1eac<br /><br />Tested successfully that the described issue has been solved.
Tested in addition that converting the parameters still works when updating from 4.4-dev.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41475">issues.joomla.org/tracker/joomla-cms/41475</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41475">issues.joomla.org/tracker/joomla-cms/41475</a>.</sub>

**Pull request merged**